// ./protoc -I . --java_out=gen --plugin=protoc-gen-grpc-java=protoc-gen-grpc-java-1.54.0-windows-x86_64.exe --grpc-java_out=gen event.proto
syntax = "proto3";

option java_multiple_files = true;
option java_package = "sr.grpc.gen";
option java_outer_classname = "EventProto";

package event_subscriber;

enum EventType{
  CONCERT = 0;
  FILM_SHOW = 1;
  FOOTBALL = 2;
  CLUB_PARTY = 3;
  FESTIVAL = 4;
  ART_SHOW = 5;

}

message SubscribeRequest {
  string country = 1;
  repeated EventType type = 2;
}

message Address{
  string country = 1;
  string city = 2;
  string zipCode = 3;
  string streetAddress = 4;
}

message Event
{
  string description = 1;
  Address address = 2;
  int32 placesNumber = 3;
  repeated EventType type = 4;
}

service EventSubscriber {
  rpc subscribe(SubscribeRequest) returns (stream Event);
}